### PR TITLE
Feature/ocbm 158 set standard

### PR DESCRIPTION
**Pull request by jayashruthi-FS**
Implemented a standardized list user interface (UI) with the following components:

1.Successfully developed a comprehensive Table UI, integrating mock data.
2.Designed and completed a Search Box UI.
3.Implemented a functional button to facilitate the creation of new fields.
4.Successfully incorporated table scroll functionality.

Figma Screenshot:
image

Implementation
image

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /// <reference types="vite/client" />
 
 export class Config {
-  static OMNEX_BACKEND_URL = import.meta.env.VITE_APP_API_BASE_URL;
```
Comment (chandran-kannan) [inline src/config/index.ts]: revert this change and use env, also don't hardcode the base url

File: src/components/reusable/table/index.tsx [hunk 2]
```diff
@@ -12,7 +12,8 @@ type customPros = {
 export function Table(props: TableProps<customPros>) {
   return (
     <>
-      <AntdTable {...props} />
```
Comment (chandran-kannan) [inline src/components/reusable/table/index.tsx]: revert the this change

File: src/pages/setStandards/SetStandardList.tsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
+import { DeleteIcon, PencilIcon, SearchIcon } from '@/assets/icons';
+import { Table } from '@/components/reusable/table';
+import { Button, Input } from '@/components';
+
+const SetStandardList = () => {
```
Comment (chandran-kannan) [inline src/pages/setStandards/SetStandardList.tsx:6]: move this constants to constants file or create `data.ts`  file in `src/pages/setStandards/SetStandardList.tsx` and move it

File: src/pages/setStandards/SetStandardList.tsx [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+  return (
+    <>
+      <div className="flex justify-between mb-7">
+        <p className="font-semibold text-2xl leading-6">CBM SET STANDARDS</p>
+        <Input
+          className="border-2 w-[340px] h-12 flex gap-3 text-sm p-4"
+          leftIcon={<SearchIcon />}
+          placeholder="Search Sensor, Machine..."
+        ></Input>
```
Comment (chandran-kannan) [inline src/pages/setStandards/SetStandardList.tsx]: self close the Input tag 
solution:
<Input
          className="border-2 w-[340px] h-12 flex gap-3 text-sm p-4"
          leftIcon={<SearchIcon />}
          placeholder="Search Sensor, Machine..."
        />

File: src/pages/setStandards/SetStandardList.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        <Button
+          className="font-medium text-base w-52 h-10 leading-4 flex justify-center items-center"
```
Comment (chandran-kannan) [inline src/pages/setStandards/SetStandardList.tsx:101]: add button `type` attribute

File: src/pages/setStandards/index.css [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  word-spacing: normal;
+}
+
+.ant-table-thead {
+  font-weight: 400;
+  font-size: 16px;
+  font-family: 'Gotham-medium';
+}
+
+table {
```
Comment (chandran-kannan) [inline src/pages/setStandards/index.css]: should not style using element selector

File: src/pages/setStandards/index.css [hunk 7]
```diff
@@ -0,0 +1,35 @@
+.operatingRange {
+  word-spacing: 1rem;
+}
+
+.thresholdValue {
+  word-spacing: 1rem;
+}
+
+.ant-table-thead > tr > th:nth-child(6) {
```
Comment (chandran-kannan) [inline src/pages/setStandards/index.css]: why  antd styles are overwritten here

File: src/pages/setStandards/SetStandardList.tsx [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+          label="+ Create Standards"
+        />
+      </div>
+      <Table
+        columns={columns}
+        dataSource={dataSource}
+        scroll={{ x: 'calc(1500px + 50%)' }}
```
Comment (chandran-kannan) [inline src/pages/setStandards/SetStandardList.tsx]: why scroll attribute added here?

**Review by chandran-kannan — CHANGES_REQUESTED** [review]

**Comment by chandran-kannan** [comment]
Hi @jayashruthi-FS add figma design screen shots in this PR

File: src/pages/setStandards/SetStandardList.tsx (hunk 8 above)
Comment (jayashruthi-FS) [reply src/pages/setStandards/SetStandardList.tsx]: image

This is why i added scroll attribute here, 
Is it ok ? or should i wrap this table in div and give width and overflow x?
CC: @Roslind-Vasantha

File: src/pages/setStandards/index.css (hunk 7 above)
Comment (jayashruthi-FS) [reply src/pages/setStandards/index.css]: Needed to customize the table. 
Now I have added classname, so that it doesn't overwrite the default style on other tables.

File: src/pages/setStandards/index.css (hunk 6 above)
Comment (jayashruthi-FS) [reply src/pages/setStandards/index.css]: I added a classname to that table.

File: src/pages/setStandards/SetStandardList.tsx (hunk 5 above)
Comment (jayashruthi-FS) [reply src/pages/setStandards/SetStandardList.tsx:101]: Added type and variant attributes in the button.

File: src/pages/setStandards/SetStandardList.tsx (hunk 4 above)
Comment (jayashruthi-FS) [reply src/pages/setStandards/SetStandardList.tsx]: Fixed it.

File: src/pages/setStandards/SetStandardList.tsx (hunk 3 above)
Comment (jayashruthi-FS) [reply src/pages/setStandards/SetStandardList.tsx:6]: Moved data source constant to a constant file.
i didn't move columns constants to the constant file because in the future table related functions will be written there.

File: src/components/reusable/table/index.tsx (hunk 2 above)
Comment (jayashruthi-FS) [reply src/components/reusable/table/index.tsx]: Reverted it

File: src/config/index.ts (hunk 1 above)
Comment (jayashruthi-FS) [reply src/config/index.ts]: Fixed it.

File: src/components/reusable/table/index.css [hunk 9]
```diff
@@ -1,8 +1,8 @@
-.ant-table-row:nth-child(even)
+.ant-table-row:nth-child(odd)
```
Comment (Jyshe) [inline src/components/reusable/table/index.css:1]: did u check the other areas where the table is used? is that working fine?

File: src/pages/setStandards/index.css [hunk 10]
```diff
@@ -0,0 +1,35 @@
+.set-table .operatingRange {
```
Comment (Jyshe) [inline src/pages/setStandards/index.css:1]: why there is an external css file? we can do only the necessary changes in index.css or follow the ANTD theme and configure.

File: src/components/reusable/table/index.css (hunk 9 above)
Comment (jayashruthi-FS) [reply src/components/reusable/table/index.css:1]: Yes checked, 
it is working fine...

File: src/pages/setStandards/index.css (hunk 10 above)
Comment (jayashruthi-FS) [reply src/pages/setStandards/index.css:1]: I needed to customize a table in a Figma design. I added a classname to ensure that it doesn't override the default style on other tables. Could you please review it to see if any modifications are needed?

**Review by chandran-kannan — APPROVED** [review]
